src/bigstring_unix.ml

```diff
@@ -3,6 +3,29 @@
 
 type t = (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t
 
+(** {2 I/O} *)
+
+let get_bounds name ?(off=0) ?len t =
+  let buffer_len = Bigarray.Array1.dim t in
+  let len = match len with
+    | Some len -> len
+    | None -> buffer_len - off
+  in
+  if len < 0 || off < 0 || buffer_len - off < len
+  then invalid_arg ("Bigstring_unix." ^ name)
+  else (off, len)
+
+external read_fd  : Unix.file_descr -> t -> int -> int -> int = "bigstring_read"
```

Comment:
is there a risk of collision there? could it be something like `ocaml_bigstring_unix_read`, which is less likely to collide with other libraries?

Reply:
I have no idea. Let's rename it as you suggest.